Sources/Parsing/Parsers/Bool.swift

```diff
@@ -19,24 +19,34 @@ extension Bool {
     .init()
   }
 
-  /// A parser that consumes a Boolean value from the beginning of a substring.
+  /// A parser that consumes a Boolean value from the beginning of a substring's UTF-8 view.
   ///
-  /// This parser only recognizes the literal "true" and "false" characters:
+  /// This overload is provided to allow the `Input` generic to be inferred when it is
+  /// `Substring.UTF8View`.
   ///
-  /// ```swift
-  /// Bool.parser().parse("true Hello"[...])  // (output: true, rest: " Hello")
-  /// Bool.parser().parse("false Hello"[...]) // (output: false, rest: " Hello")
-  /// Bool.parser().parse("1 Hello"[...])     // (output: nil, rest: "1 Hello")
-  /// ```
+  /// - Parameter inputType: The `Substring.UTF8View` type. This parameter is included to mirror the
+  ///   interface that parses any collection of UTF-8 code units.
+  /// - Returns: A parser that consumes a Boolean value from the beginning of a substring's UTF-8
+  ///   view.
+  @inlinable
+  public static func parser(
+    of inputType: Substring.UTF8View.Type = Substring.UTF8View.self
+  ) -> Parsers.BoolParser<Substring.UTF8View> {
+    .init()
+  }
+
+  /// A parser that consumes a Boolean value from the beginning of a substring.
+  ///
+  /// This overload is provided to allow the `Input` generic to be inferred when it is `Substring`.
   ///
-  /// - Parameter inputType: The substring type. This parameter is included to mirror the interface
-  ///   that parses UTF-8 code units.
+  /// - Parameter inputType: The `Substring` type. This parameter is included to mirror the
+  ///   interface that parses any collection of UTF-8 code units.
   /// - Returns: A parser that consumes a Boolean value from the beginning of a substring.
   @inlinable
   public static func parser(
     of inputType: Substring.Type = Substring.self
-  ) -> Parsers.SubstringBoolParser {
-    .init()
+  ) -> FromUTF8View<Parsers.BoolParser<Substring.UTF8View>> {
+    .init(.init())
   }
```

Comment:
We already had many overloads for `Substring`, but instead of ad hoc parsers for each instance, we can now use a single `FromUTF8View` wrapper instead.